### ISSUE TITLE
[FR] regular expressions support

### ISSUE THREAD

**Issue by beelze** [dalance/procs#554]
`procs` has a well-developed and consistent search logic that covers many use cases. Though, it is not rare when "grepping" may be more familiar and convenient. But because of watching mode and headers, `grep` is not suitable.

The idea is to add __new regex filtering mode__, leave current numeric/textual search as is. The reasons are:
1. **Regexes may be easier to write.** As I said, for those who often "greps" logs (for instance) regexes are familiar, powerful and flexible way of filtering. And `procs` output is very similar to logs. *I've come to this when I tried to watch for spawning and killing instances of two binaries while messing with processed like man or emacs which has same strings in command lines. A lot of `--and` and `--or`  options plus adding spaces did the trick but it took some time. And here I thought "It would be nice to did the same with a simple regex"*
2. **Regexes are flexible.** For instance, watching for some processes by state and name could be tricky but (assuming the string we're testing with regex is columns joined with space) I can easily write something like `( S .*Rambox| D .*firefox)`. Currently I need to check my config for `Exact` or `Partial` and think about `--and` and `--or` priorities. 
3. Of course, it is opinion-based, but for most complex cases **regexes are shorter and more predictable** comparing to combination of logical conditions.

Desired implementation:
1. Switching modes options: `--text` (current search mode), `--regex` (regex mode) and `--smart` (regex autodetection by specific patterns, like |  (…), * ? etc)
2. In regex mode, instead if `[KEYWORD]…`, arguments there is a single `<REGEX> argument
3. Interactive regex-based filtering in watch mode (as for me, this feature can save a lot of time for a complex filtering cases)

**Comment by sabbellasri** [comment]
Hi Beelze, I am interested in this issue. Could you please assign it to me?

**Comment by beelze** [comment]
> Hi Beelze, I am interested in this issue. Could you please assign it to me?

Sorry, but I have no permissions for that. Please ask @dalance instead.

**Comment by dalance** [comment]
@sabbellasri Thank you for your interesting in this issue.

**Comment by beelze** [comment]
> Yeah, I am working on it and will update u with the code by weekend sorry for the late reply

Just tried, great! But I noticed lookarounds aren't supported – perhaps because of regex crate limitation? I've found [fancy-regex](https://github.com/fancy-regex/fancy-regex?tab=readme-ov-file) supports lookarounds and other "fancy" syntax but _"delegates as much as possible to the NFA engine"_.

**Comment by sabbellasri** [comment]
Yeah, It seems lookarounds aren’t supported because the project currently relies on Rust’s regex crate, which doesn’t support them.
One possible approach could be to integrate something like fancy-regex, which supports lookarounds while still delegating simpler patterns to the faster NFA engine.
Would you be open to exploring this direction? I can take a look and try implementing it if it aligns with the project goals

**Comment by beelze** [comment]
> Would you be open to exploring this direction?
Yes.

**Comment by sabbellasri** [comment]
Thanks for confirming!

I’ll explore integrating `fancy-regex` to add support for lookarounds while ensuring simpler patterns still leverage the faster `regex` crate. I’ll work on this and open a new PR with the proposed changes by next week.
